manifests/params.pp

```diff
@@ -176,7 +176,11 @@
   if defined('$trusted') and $::trusted['authenticated'] == 'local' {
     $puppetmaster            = undef
   } else {
-    $puppetmaster            = $::puppetmaster
+    if defined('$::puppetmaster') {
```

Comment:
As defined() requires Puppet 3.6 to return true, this will need to be mandatory or the code will stop working on older versions.

Change metadata.json to require 3.6 and remove the obsolete trusted check above, else you'd have to introduce version checks.